### ISSUE TITLE
Spectating games

### ISSUE THREAD

**Issue by SirPython** [Cardshifter/HTML-Client#95]
Seeing that a `ServerStatusMessage` includes the games that are currently running, I thought it would be interesting to utilize this information in creating a feature where you can spectate currently running games.

For example, there might be a button in the lobby that says, "Spectate", which would then bring you to a screen where you are presented with a list of all running games, and the users in the game. After choosing a game, you are launched into a screen where you can watch the game as it is being played.

**Comment by Zomis** [comment]
yes, this is planned. See also https://github.com/Cardshifter/Cardshifter/issues/110

**Comment by Phrancis** [comment]
This would involve a couple different things, including connecting the would-be spectator to the other players' WebSocket connection, then generating the game from all the events. It may be quite difficult to do unless a spectator joins at the very beginning of the game, or we make changes to the server to accept a spectator connection and regenerate the game from scratch for the spectator.

This needs some discussion but may be a possibility in the future.

**Comment by Zomis** [comment]
It is (or at least was) possible in Royal Game of Ur, so it should be possible in Cardshifter as well.
